corehq/apps/ota/views.py

```diff
@@ -44,7 +43,7 @@
 
 @location_safe
 @handle_401_response
-@login_or_digest_or_basic_or_apikey_or_token()
+@login_or_digest_or_basic_or_apikey_or_token(default=BASIC)
```

Comment:
I believe the problem is that `@login_or_digest_or_basic_or_apikey_or_token(default=BASIC)` cannot work as described.  You can only authenticate with digest auth if it is the default when no information is provided by the client.

We could remove the default argument and make a second decorator that explicitly doesn't support digest.

Reply:
Is this simply a question of a misleadingly named function, or do you think this breaks some existing functionality. If its the former i am happy to fix in a follow up, but I want to make sure this gets in before i head off to bed, unless you want to take it over. ICDS really needs this fixed tonight.

If you believe this will break something, then i am not sure i understand. The default will be digest except for this view, which i think is only hit by mobile and therefore we want to force basic auth.

Reply:
Cal, Farid, and I offlined about this extensively and I think we're all on the same page now, but for the sake of this PR's history:

I think it's more of a naming question (although with issues beyond just code clarity), and agree that fixing ICDS is much higher a priority than getting this perfect.